src/fastmeteo/data/arpege.py

```diff
@@ -0,0 +1,113 @@
+import tempfile
+from datetime import datetime, timedelta, timezone
+from io import BytesIO
+from pathlib import Path
+from typing import Any
+
+import httpx
+import numpy as np
+import pandas as pd
+import xarray as xr
+from tqdm.auto import tqdm
+
+from .. import aero
+from ..core.grid import Grid
+
+tempdir = Path(tempfile.gettempdir())
+
+bare_url = "https://object.data.gouv.fr/meteofrance-pnt/pnt/"
+
+
+def download_with_progress(url: str) -> BytesIO:
+    with httpx.stream("GET", url) as r:
+        total_size = int(r.headers.get("Content-Length", 0))
+        buffer = BytesIO()
+        with tqdm(
+            total=total_size, unit="B", unit_scale=True, desc=url.split("/")[-1]
+        ) as progress_bar:
+            for chunk in r.iter_bytes():
+                buffer.write(chunk)
+                progress_bar.update(len(chunk))
+        buffer.seek(0)
+        return buffer
+
+
+class Arpege(Grid):
+    """
+    Class to handle the ARPEGE data.
+    """
+
+    def __init__(self, local_store: str) -> None:
+        self.local_store = local_store
+        self.model = "ARPEGE"
+        self.grib = "025"
+        self.package = "IP1"
+        self.time_range = "000H024H"
+        self.run_date = [0, 6, 12, 18]
+        self.features = ["r", "t", "u", "v", "z"]
+
+    def get_latest_run_time(self, time: np.datetime64) -> datetime:
+        utc_now = pd.to_datetime(time).tz_localize("UTC")
+        candidate = datetime(
+            utc_now.year,
+            utc_now.month,
+            utc_now.day,
+            utc_now.hour,
+            tzinfo=timezone.utc,
+        )
+        run_time = datetime(
+            candidate.year,
+            candidate.month,
+            candidate.day,
+            tzinfo=timezone.utc,
+        )
+        for hour in self.run_date:
```

Comment:
For the moment, this is not great. We are not trying to get all the predictions for a given timestamp, but rather the first available prediction for a given timestamp.